service_msgs/msg/ServiceEventInfo.msg

```diff
@@ -0,0 +1,18 @@
+uint8 REQUEST_SENT = 0
+uint8 REQUEST_RECEIVED = 1
+uint8 RESPONSE_SENT = 2
+uint8 RESPONSE_RECEIVED = 3
+
+# The type of event this message represents
+uint8 event_type
+
+# Timestamp for when the event occurred (sent or received time)
+builtin_interfaces/Time stamp
+
+# Unique identifier for the client that sent the service request
+# Note, this is only unique for the current session
+unique_identifier_msgs/UUID client_id
```

Comment:
Considering the discussion in https://github.com/ros2/rmw/pull/328, it's possible this identifier coming from the middleware may be more (or less) than 16 bytes (ie. it is not actually a UUID). So we probably want to update this to be an array of [24 bytes](https://github.com/ros2/rmw/blob/67b0f0a229c0bd665913a1d937ee82bf896ad554/rmw/include/rmw/types.h#L42), which is the current maximum and leave a comment where this is coming from.

Reply:
I think all implementations use 16 bytes (128 bits).
I'm not sure why 24 bytes was chosen in rmw.

Reply:
Jacob and I discussed this before he left for his travels, but regardless of the size of this field, using UUID for it, is not correct, in my opinion, if the `client_id` is the entity ID from DDS/rmw. They are GID/GUID and are not guaranteed to be "universally" unique like a UUID.

So we should use a different type for this.

However, if I'm wrong and `client_id` is something we're generating at the ROS level and it is a UUID this is ok, but I don't think that's the case off-hand.

Reply:
Sounds good!

Should I create a new message type called GID? is that name fine?
Where should I create it? is builtin_interfaces a good place? in a new package?
Do we need a new message type? Maybe I can use an array directly in `ServiceEventInfo`.

Reply:
After a lot of discussion, we did in fact come to conclusion that this should be a GID (as defined in the DDS spec), rather than a UUID.  They both happen to be 16-bytes long, but they are semantically different (and DDS assigns certain meaning to certain bytes).

I attempted to reuse the GID message from https://github.com/ros2/rmw_dds_common/blob/rolling/rmw_dds_common/msg/Gid.msg , but I ran into circular dependency issues between this repository and rmw_dds_common.  So what I ended up doing here was to call the field client_gid, and make it a 16-byte char field.  This also required a few other changes in some other repositories, and that is now all pushed.

So I think we can call this particular conversation resolved.  I'm going to mark it as such, kick off CI on the whole lot of PRs, and see where we are.